kola/tests/bpf/local-gadget.go

```diff
@@ -0,0 +1,160 @@
+  files:
+    - path: /opt/local-gadget.tar.gz
+      filesystem: root
+      mode: 0755
+      contents:
+        remote:
+          url: https://github.com/kinvolk/inspektor-gadget/releases/download/v{{ .Version }}/local-gadget-linux-{{ .Arch }}.tar.gz
+          verification:
+            hash:
+              function: sha512
+              sum: {{ .Sum }}
+    - path: /opt/local-gadget.cmd
+      filesystem: root
+      mode: 0644
+      contents:
+        inline: |
+          {{ .Cmd }}
+systemd:
+  units:
+    - name: prepare-local-gadget.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Unpack local-gadget to /opt/bin/
+        ConditionPathExists=!/opt/bin/local-gadget
+        [Service]
+        Type=oneshot
+        RemainAfterExit=true
+        Restart=on-failure
+        ExecStartPre=/usr/bin/mkdir --parents /opt/bin
+        ExecStartPre=/usr/bin/tar -v --extract --file /opt/local-gadget.tar.gz --directory /opt/bin --no-same-owner
+        ExecStart=/usr/bin/rm /opt/local-gadget.tar.gz
+        [Install]
+        WantedBy=multi-user.target
+    - name: local-gadget.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Run local-gadget
+        After=prepare-local-gadget.service
+        Requires=prepare-local-gadget.service
+        [Service]
+        User=root
+        Type=fork
+        RemainAfterExit=true
+        Restart=on-failure
+        ExecStart=/opt/bin/local-gadget
+        StandardInput=file:/opt/local-gadget.cmd
+        StandardOutput=file:/tmp/local-gadget.res
+        [Install]
+        WantedBy=multi-user.target`
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:     localGadgetTest,
+		Name:    `bpf.local-gadget`,
+		Distros: []string{"cl"},
+		// required while SELinux policy is not correcly updated to support
+		// `bpf` and `perfmon` permission.
+		Flags: []register.Flag{register.NoEnableSelinux},
+	})
+}
+
+func localGadgetTest(c cluster.TestCluster) {
+	arch := strings.SplitN(kola.QEMUOptions.Board, "-", 2)[0]
+	gadget := binaries[arch]
+
+	gadget.Arch = arch
+
+	tmpl, err := template.New("user-data").Parse(config)
+	if err != nil {
+		c.Fatalf("parsing user-data: %w", err)
+	}
+
+	c.Run("dns gadget", func(c cluster.TestCluster) {
+		gadget.Cmd = `create dns kola-dns --container-selector=shell01
+          list-traces
+          stream -f kola-dns`
+
+		var buf bytes.Buffer
+		if err := tmpl.Execute(&buf, gadget); err != nil {
+			c.Fatalf("rendering user-data: %w", err)
+		}
+
+		node, err := c.NewMachine(conf.ContainerLinuxConfig(buf.String()))
+		if err != nil {
+			c.Fatalf("creating node: %v", err)
+		}
+
+		docker.GenDockerImage(c, node, "dig", []string{"dig"})
+
+		if _, err := c.SSH(node, "docker run --rm --name shell01 dig dig flatcar-linux.org"); err != nil {
+			c.Fatalf("unable to run docker cmd: %v", err)
+		}
+
+		out, err := c.SSH(node, "cat /tmp/local-gadget.res  | grep pkt_type | head --lines 1 | jq -r .name")
```

Comment:
```suggestion
		out, err := c.SSH(node, "grep -m 1 pkt_type /tmp/local-gadget.res | jq -r .name")
```
This makes the processes run without error, not a real problem but if it gets copied later where pipefail is set, grep will fail when head terminates earlier because it does not consume the full input

Reply:
Thanks for the tip.